cesm_tools.py

```diff
@@ -221,6 +223,18 @@ def setup_cesm_explicit(
             cyclic_x=cyclic_x,
         )
 
+        # Load Balancing Math
+        total_number_of_points = nx * ny
+        nodes=1
+        pts_per_processor = total_number_of_points/ float(processors_per_node)
+        while not pts_per_processor < ideal_number_of_points_per_core_ceiling:
+            pts_per_processor = total_number_of_points / float(processors_per_node * nodes)
+            nodes = nodes+1
```

Comment:
I think you want to set `nodes = nodes+1` before updating `pts_per_processor`. For example, if your grid has 900*128 = 115200 cells, your logic currently is:

(line 229) `pts_per_processor = 115200 / 128 = 900`
(line 230) `not 900 < 800` so enter the loop
(line 231) `nodes` is still 1, so `pts_per_processor` is still 900
(line 232) `nodes = 2`
(line 230) `not 900 < 800` so enter the loop
(line 231) `nodes` is 2, so `pts_per_processor` is 450
(line 232) `nodes = 3`
(line 230) `450 < 800` so do not enter the loop
(line 237) `nodes = 3`, so we're requesting 384 tasks for the ocean (but we'd be happy with 256 since that would end up with `pts_per_processor = 450`)

Swapping these two lines would be

(line 229) `pts_per_processor = 115200 / 128 = 900`
(line 230) `not 900 < 800` so enter the loop
(line 231) `nodes = 2`
(line 232) `nodes` is 2, so `pts_per_processor` is 450
(line 230) `450 < 800` so do not enter the loop
(line 237) `nodes = 2`, so we're requesting 256 tasks for the ocean

Reply:
Hey Mike,

Shoot that was a silly mistake. I'll clean that up.

Thanks,
Manish